Podverse/PodcastsTableViewController.swift

```diff
@@ -285,28 +297,32 @@ extension PodcastsTableViewController {
         super.podcastDeleted(notification)
         
         if let feedUrl = notification.userInfo?["feedUrl"] as? String, let index = self.subscribedPodcastsArray.index(where: { $0.feedUrl == feedUrl }) {
+            self.subscribedPodcastsArray.remove(at: index)
+
             DispatchQueue.main.async {
-                self.subscribedPodcastsArray.remove(at: index)
                 self.tableView.deleteRows(at: [IndexPath(row: index, section: 0)], with: .fade)
             }
         }
     }
     
     func refreshParsingStatus(_ notification:Notification) {
-        let total = self.parsingPodcastsList.urls.count
-        let currentItem = self.parsingPodcastsList.currentlyParsingItem
+        let total = self.parsingPodcasts.urls.count
+        let currentItem = self.parsingPodcasts.currentlyParsingItem
         
-        DispatchQueue.main.async {
-            if total > 0 && currentItem < total {
+        if total > 0 && currentItem < total {
```

Comment:
Just curious, why is this different that dispatching once and having the if statements inside the dispatch?

Reply:
I had this notion that, if you dispatch to the main queue, you should only dispatch the parts that will directly change the UI, whereas evaluating that conditional logic wouldn't directly change the UI, so I thought that should be kept off the main thread.

It seems like I'm splitting hairs here though. I'll go ahead and move the conditional into the main thread.